test-network-function/lifecycle/suite.go

```diff
@@ -135,7 +143,11 @@ func restoreDeployments(env *config.TestEnvironment) {
 
 		if len(notReadyDeployments) > 0 {
 			// Wait until the deployment is ready
-			waitForAllDeploymentsReady(deployment.Namespace, scalingTimeout, scalingPollingPeriod)
+			notReady := waitForAllDeploymentsReady(deployment.Namespace, scalingTimeout, scalingPollingPeriod)
+			if notReady != 0 {
+				collectNodeAndPendingPodInfo(deployment.Namespace)
```

Comment:
@greyerof I think we missed this? Aborting here would make the next if block useless. With your original design, we should only do this after the last attempt of restoring the replicas?

Reply:
If I'm not wrong, in both the original and current restoreDeployments, the next "if block" won't be reached if waitForAllDeploymentsReady on line 146 fails. The difference is that in the original version, it was implemented with  gomega.Eventually, which makes the TC to fail but the TS continues, while this log.Fatalf() will stop the whole TS/tnf.